### PR TITLE
[Network, Redis, Monitor] Linter fixes and more~!

### LINKED ISSUE #5593: [Redis] list-all command should be folded into the list command
The list-all command should be deprecated and the subscription vs. resource-group functionality should be folded into the existing list command like the rest of the list operations in the CLI.

---

### Environment summary

Install Method (e.g. pip, interactive script, apt-get, Docker, MSI, edge build) / CLI version (`az --version`) / OS version / Shell Type (e.g. bash, cmd.exe, Bash on Windows)  

```
azure-cli (2.0.28)

acr (2.0.21)
acs (2.0.26)
advisor (0.1.2)
appservice (0.1.26)
backup (1.0.6)
batch (3.1.10)
batchai (0.1.5)
billing (0.1.7)
cdn (0.0.13)
cloud (2.0.12)
cognitiveservices (0.1.10)
configure (2.0.14)
consumption (0.2.1)
container (0.1.18)
core (2.0.28)
cosmosdb (0.1.19)
dev-tools (0.1.1)
dla (0.0.18)
dls (0.0.19)
eventgrid (0.1.10)
extension (0.0.9)
feedback (2.1.0)
find (0.2.8)
interactive (0.3.16)
iot (0.1.17)
keyvault (2.0.19)
lab (0.0.17)
monitor (0.1.2)
network (2.0.23)
nspkg (3.0.1)
profile (2.0.19)
rdbms (0.0.12)
redis (0.2.11)
reservations (0.1.1)
resource (2.0.24)
role (2.0.19)
servicefabric (0.0.10)
sql (2.0.21)
storage (2.0.25)
testsdk (0.1.0)
vm (2.0.27)

Python location '/Users/travisprescott/github/azure-cli/env/bin/python'
Extensions directory '/Users/travisprescott/.azure/cliextensions'

Python (Darwin) 2.7.12 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:43:17) 
[GCC 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2336.11.00)]
```

### PR DESCRIPTION
**Pull request by tjprescott**
**Network**

Removes `--ids` from all list commands except `vnet subnet` and `vnet peering`, which have too many recent uses to remove.  They will be deprecated and removed as part of #6062. This is technically a breaking change.

**Monitor**

Addresses linter fixes. No breaking changes.

**Redis**

Addresses linter fixes. Closes #5593 by folding the `list-all` functionality into the `list` command.  Deprecates the `list-all` and `import-method` commands in favor of `list` and `import` respectively. No breaking changes (at least in this release).

**Linter Script**

Changes the output format to be easier to read and interpret. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6065
This is an **experimental preview** for @microsoft users.

**Comment by tjprescott** [comment]
@williexu please take a look at the changes since your approval.

File: tools/automation/cli_linter/linter.py [hunk 1]
```diff
@@ -113,30 +114,36 @@ def run(self, run_params=None, run_commands=None, run_command_groups=None, run_h
...
         if run_commands and self._rules.get('commands'):
-            print(os.linesep + 'Running Linter on Commands:')
             self._run_rules('commands')
 
         if run_params and self._rules.get('params'):
-            print(os.linesep + 'Running Linter on Parameters:')
             self._run_rules('params')
 
+        if not self.exit_code:
+            print('\nNo violations found.')
```
Comment (williexu) [inline tools/automation/cli_linter/linter.py]: Use os.linesep.

File: tools/automation/cli_linter/linter.py [hunk 2]
```diff
@@ -113,30 +114,36 @@ def run(self, run_params=None, run_commands=None, run_command_groups=None, run_h
...
+        colorama.deinit()
         return self.exit_code
 
     def _run_rules(self, rule_group):
-        for callable_rule in self._rules.get(rule_group):
-            for violation_msg in sorted(callable_rule()):
-                print(violation_msg)
-            print()
+        from colorama import Fore
```
Comment (williexu) [inline tools/automation/cli_linter/linter.py]: Why import again when `colorama.Fore` can be used?

File: tools/automation/cli_linter/linter.py [hunk 3]
```diff
@@ -113,30 +114,36 @@ def run(self, run_params=None, run_commands=None, run_command_groups=None, run_h
...
+        for rule_name, rule_func in self._rules.get(rule_group).items():
+            violations = sorted(rule_func()) or []
+            if violations:
+                print('- {} FAIL{}: {}'.format(Fore.RED, Fore.RESET, rule_name))
+                for violation_msg in violations:
+                    print(violation_msg)
+                print()
+            else:
+                print('- {} pass{}: {} '.format(Fore.GREEN, Fore.RESET, rule_name))
```
Comment (williexu) [inline tools/automation/cli_linter/linter.py]: kinda nitpicky but maybe only one or zero spaces? or just get rid of the string, "-  ", before the `pass/FAIL`. Feel free to ignore as well.

**Pull request merged**